oci-unit-tests/kafka_test.sh

```diff
@@ -0,0 +1,186 @@
+# shellcheck shell=dash
+
+# shellcheck disable=SC1090
+. "$(dirname "$0")/helper/test_helper.sh"
+. "$(dirname "$0")/helper/common_vars.sh"
+
+# cheat sheet:
+#  assertTrue $?
+#  assertEquals ["explanation"] 1 2
+#  oneTimeSetUp()
+#  oneTimeTearDown()
+#  setUp() - run before each test
+#  tearDown() - run after each test
+
+readonly zookeeper_port=2181
+readonly broker_port=9092
+
+oneTimeSetUp() {
+    id=$$
+
+    # Remove image before test.
+    remove_current_image
+
+    # Make sure we're using the latest OCI image.
+    docker pull --quiet "${DOCKER_IMAGE}" > /dev/null 2>&1
+
+    # Cleanup stale resources
+    tearDown
+    oneTimeTearDown
+
+    # Setup network
+    docker network create "${DOCKER_NETWORK}" > /dev/null 2>&1
+}
+
+oneTimeTearDown() {
+    docker network rm "${DOCKER_NETWORK}" > /dev/null 2>&1
+}
+
+tearDown() {
+    local container
+
+    for container in $(docker ps --filter "name=_test_${id}" --format "{{.Names}}"); do
+        debug "Removing container ${container}"
+        stop_container_sync "${container}"
+    done
+
+    for container in $(docker ps --filter "name=${DOCKER_PREFIX}" --format "{{.Names}}"); do
+        debug "Removing container ${container}"
+        stop_container_sync "${container}"
+    done
+}
+
+# Helper function to execute zookeeper with some common arguments.
+docker_run_zookeeper() {
+    docker run \
+     --network "$DOCKER_NETWORK" \
+     --rm \
+     -d \
+     -p "${zookeeper_port}":"${zookeeper_port}" \
+     --name zookeeper_test_${id} \
```

Comment:
I think that if you prefix the container name with `${DOCKER_PREFIX}` then you won't need to loop twice in the `tearDown` function.